### PR TITLE
Test pyqt6

### LINKED ISSUE #9: smoke test
add a simple function to "smoke test" packages by importing them from the package directory.
caveat: add_dll_directory also needs to be present here

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes https://github.com/open-vcpkg/python-registry/issues/8
Fixes #9

### CODE REVIEW

**Comment by Neumann-A** [comment]
image

I think we need to delete something but the questions is what? Is something collecting a huge amount of  logs/artifacts?

**Comment by Thomas1664** [comment]
> I think we need to delete something but the questions is what? Is something collecting a huge amount of logs/artifacts?

I think the binary cache accumulated many entries from every PR run

**Comment by Neumann-A** [comment]
> Why do you only have 0,5 GB? I have 2 GB

That seems to be the difference between Free and Pro. My personal account also has 2 GB but the organization has only 0,5 GB updating that costs 4 $ per user per month which is kind of strange unless the GB size also increases with the number of users. I wonder what "user" means here in organization context.

**Comment by m-kuhn** [comment]
I understood "public packages are free"
https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages#about-billing-for-github-packages

**Comment by m-kuhn** [comment]
I enabled "public" packages now in https://github.com/organizations/open-vcpkg/settings/packages

**Comment by Neumann-A** [comment]
If the same packages for users apply to orgs the package stuff needs to be moved to somewhere else in the future. (E.g. there can only be one repo with vcpkg packages.) For example I have a private repo called `my-vcpkg-packages` which just handles the package stuff for all other repos due to that limitation

**Comment by m-kuhn** [comment]
> If the same packages for users apply to orgs the package stuff needs to be moved to somewhere else in the future. (E.g. there can only be one repo with vcpkg packages.)

why?

**Comment by Neumann-A** [comment]
> why?

Strange limitation of github if i remember correctly. You'll get strange errors if another repo tries to push github packages.

**Comment by m-kuhn** [comment]
> > why?
> 
> Strange limitation of github if i remember correctly. You'll get strange errors if another repo tries to push github packages.

It might be related to packages which are linked to a repository?
Reading about this, github suggests linking packages to a repository for ACL reasons. As I think we don't care and want everything public, this should be a non issue. If this is really the reason for the aforementioned limitation.

File: ports/py-pyqt6/portfile.cmake [hunk 1]
```diff
@@ -7,91 +8,59 @@ vcpkg_from_pythonhosted(
...
-  "--no-make"
-  "--disable" "QtDesigner"
-  "--pep484-pyi"
-  "--debug"
-  "--target-dir" "${CURRENT_PACKAGES_DIR}/lib/python${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}/site-packages"
-  "--build-dir" "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel-build"
-)
+if(CMAKE_HOST_WIN32)
+    set(EXE_SUFFIX ".exe")
+endif()
```
Comment (Neumann-A) [inline ports/py-pyqt6/portfile.cmake:13]:
```suggestion
```

File: ports/py-pyqt6/portfile.cmake [hunk 2]
```diff
@@ -7,91 +8,59 @@ vcpkg_from_pythonhosted(
...
 
 set(SIPBUILD_ARGS
   "--confirm-license"
-  "--qmake" "${CURRENT_INSTALLED_DIR}/tools/Qt6/bin/qmake.exe" #TODO: append .exe only on windows
+  "--qmake" "${CURRENT_INSTALLED_DIR}/tools/Qt6/bin/qmake${EXE_SUFFIX}"
```
Comment (Neumann-A) [inline ports/py-pyqt6/portfile.cmake:17]:
```suggestion
  "--qmake" "${CURRENT_HOST_INSTALLED_DIR}/tools/Qt6/bin/qmake${VCPKG_HOST_EXECUTABLE_SUFFIX}"
```

Comment (Neumann-A) [reply ports/py-pyqt6/portfile.cmake:17]: (could require feeding a different qt.conf?)

File: ports/py-pyqt6/portfile.cmake [hunk 3]
```diff
@@ -7,91 +8,59 @@ vcpkg_from_pythonhosted(
...
-if (CMAKE_HOST_WIN32)
-    vcpkg_add_to_path(PREPEND "${SCRIPTS}/buildsystems/make_wrapper")
-    if(NOT DEFINED Z_VCPKG_MAKE)
-        vcpkg_acquire_msys(MSYS_ROOT)
-        find_program(Z_VCPKG_MAKE make PATHS "${MSYS_ROOT}/usr/bin" NO_DEFAULT_PATH REQUIRED)
-    endif()
-    set(make_command "${Z_VCPKG_MAKE}")
-    vcpkg_list(SET make_opts -j ${VCPKG_CONCURRENCY} --trace -f ${MAKEFILE})
+# Make sure sipbuild can find QtCore.dll
+vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}/tools/Qt6/bin/" "${CURRENT_INSTALLED_DIR}/bin")
```
Comment (Neumann-A) [inline ports/py-pyqt6/portfile.cmake:33]: `_HOST_` ?

File: ports/py-pyqt6/portfile.cmake [hunk 4]
```diff
@@ -7,91 +8,59 @@ vcpkg_from_pythonhosted(
...
+    COMMAND "${PYTHON3}" "-m" "sipbuild.tools.build" ${SIPBUILD_ARGS}
+    WORKING_DIRECTORY "${SOURCE_PATH}"
+    LOGNAME "sipbuild-${TARGET_TRIPLET}"
+)
+message(STATUS "Running sipbuild...finished.")
 
-else()
-    if(VCPKG_HOST_IS_OPENBSD)
-        find_program(Z_VCPKG_MAKE gmake REQUIRED)
+if(CMAKE_HOST_WIN32)
```
[the remark this answers is not in the document]
Comment (Neumann-A) [reply ports/py-pyqt6/portfile.cmake:42]: nevermind `vcpkg-make` just pull it in via an overlay.

File: ports/py-pyqt6/portfile.cmake (hunk 3 above)
Comment (m-kuhn) [reply ports/py-pyqt6/portfile.cmake:30]: Will need to check again if that was for running host tools or to discover target dependencies. For the latter case it's probably the wrong approach.

File: ports/py-pyqt6/portfile.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (m-kuhn) [reply ports/py-pyqt6/portfile.cmake:39]: It's from `vcpkg_qmake_build.cmake` which is written for qt5. I guess `vcpkg-qmake` is the port to use here, I didn't find `vcpkg-make`, where does that live?

[the remark this answers is not in the document]
Comment (Neumann-A) [reply ports/py-pyqt6/portfile.cmake:39]: https://github.com/microsoft/vcpkg/pull/33198

[the remark this answers is not in the document]
Comment (m-kuhn) [reply ports/py-pyqt6/portfile.cmake:39]: Nice port!
Pain points encountered so far:

- `z_vcpkg_make_get_cmake_vars()` will flush the {triplet}-{build-type-short} (x64-windows-rel) build folder. Bandaid: manually call this before running the sipbuild command that triggers qmake to prepare the build directory. Thanks for adding a cache for the cmake vars!
- the generated Makefile isn't make compatible (jom works fine), do we need a `vcpkg-nmake` version of this with `PREFER_JOM`?
```
/bin/sh: -c: line 2: syntax error: unexpected end of file
make: *** [Makefile:91: sub-qmlscene-all-ordered] Error 2
```

[the remark this answers is not in the document]
Comment (Neumann-A) [reply ports/py-pyqt6/portfile.cmake:39]: > the generated Makefile isn't make compatible (jom works fine), do we need a vcpkg-nmake version of this with PREFER_JOM?

If that is the case just use `vcpkg-qmake`.

**Comment by m-kuhn** [comment]
Thanks for the add_dll_directory patch @Neumann-A https://github.com/open-vcpkg/python-registry/pull/3/commits/c38149fb3cb6e05012bcca479e39df8d69618983

**Comment by Neumann-A** [comment]
> Thanks for the add_dll_directory patch @Neumann-A https://github.com/open-vcpkg/python-registry/commit/c38149fb3cb6e05012bcca479e39df8d69618983

No problem more or less stolen from conda ;) 

But be aware that embedded python needs a slightly better approach: The path  `const wchar_t python_path[] = L"\\tools\\python3";` can be `\\tools\\<anything>` in reality. Currently the patch only enables python3 itself to find the dlls. Stuff embedding python3 will still not be able to find the dlls without additional setup.

**Comment by m-kuhn** [comment]
> 
> But be aware that embedded python needs a slightly better approach: The path `const wchar_t python_path[] = L"\\tools\\python3";` can be `\\tools\\<anything>` in reality. Currently the patch only enables python3 itself to find the dlls. Stuff embedding python3 will still not be able to find the dlls without additional setup.

Sounds like this should be done when loading the dll instead ideally?

Any last comments from your side before merging this?

**Comment by m-kuhn** [comment]
I'll merge later today unless there's an additional comment coming.
I'm quite happy with what's here. Even if it's not perfect yet, it's a good starting point for future improvements.

**Pull request merged**